utils/tests/test_search_general.py

```diff
@@ -0,0 +1,194 @@
+        # additional spaces, which is not a problem.
+        self.assertIn(expected_filter_query_split[0]['name'], filter_query_split[0]['name'])
+        self.assertIn(expected_filter_query_split[0]['remove_url'], filter_query_split[0]['remove_url'])
+
+    def test_split_filter_query_cluster_facet(self):
+        # We check that the combination of a duration filter, a facet filter (CC Attribution) and a cluster filter
+        # works correctly.
+        filter_query_string = u'duration:[0 TO 10] license:"Attribution"'
+        filter_query_split = split_filter_query(filter_query_string, '1')
+
+        expected_filter_query_split = [
+            {'remove_url': u'duration:[0 TO 10]', 'name': u'license:Attribution'}, 
+            {'remove_url': u'duration:[0 TO 10] license:"Attribution"', 'name': 'Cluster #1'}
+        ]
+
+        # check that the cluster facet exists
+        filter_query_names = [filter_query_dict['name'] for filter_query_dict in expected_filter_query_split]
+        self.assertIn('Cluster #1', filter_query_names)
+
+        # the order does not matter for the list of facet dicts.
+        # we get the index of the correspondings facets dicts.
+        cc_attribution_facet_dict_idx = filter_query_names.index('license:Attribution')
+        cluster_facet_dict_idx = filter_query_names.index('Cluster #1')
+
+        self.assertIn(expected_filter_query_split[0]['name'], 
+                      filter_query_split[cc_attribution_facet_dict_idx]['name'])
+        self.assertIn(expected_filter_query_split[0]['remove_url'], 
+                      filter_query_split[cc_attribution_facet_dict_idx]['remove_url'])
+
+        self.assertIn(expected_filter_query_split[1]['name'], 
+                      filter_query_split[cluster_facet_dict_idx]['name'])
+        self.assertIn(expected_filter_query_split[1]['remove_url'], 
+                      filter_query_split[cluster_facet_dict_idx]['remove_url'])
+
+    def test_search_prepare_query(self):
+        # we test that some query parameters get correctly setted in the Solr query object.
+        query_params = {
+            'search_query': u'cat',
+            'filter_query': u'duration:[1 TO 10] is_geotagged:1',
+            'sort': [u'duration desc'],
+            'current_page': 1,
+            'sounds_per_page': settings.SOUNDS_PER_PAGE,
+            'id_weight': 10,
+            'tag_weight': 0,
+            'description_weight': 0,
+            'username_weight': 0,
+            'pack_tokenized_weight': 0,
+            'original_filename_weight': 0,
+            'grouping': True,
+            'in_ids': [],
+        }
+
+        query = search_prepare_query(**query_params)
+
+        self.assertEqual(query.params['q'], 'cat')
+        self.assertEqual(query.params['qf'], 'id^10')
+        self.assertEqual(query.params['fq'], 'duration:[1 TO 10] is_geotagged:1')
+        self.assertTrue(query.params['group'])
```

Comment:
why not testing the rest of the parameters?

Reply:
There are many parameters in the query.params dict and I am not sure I understand all of them.
Shall I copy/paste the dictionary obtained when executing the function manually, and just use `assertDictEqual()` for comparing at once the expected (copied) parameters and the resulting ones?